file.go

```diff
@@ -90,14 +92,14 @@ func ServeFile(server DownloadServer, contentType, path string) error {
 		return err
 	}
 	if info.IsDir() {
-		return errors.Errorf("invalid path %s", path)
+		return fmt.Errorf("invalid path %s", path)
 	}
-	return ServeContent(server, file, contentType, info.Name(), info.ModTime(), info.Size())
+	return serveContent(server, file, contentType, info.Name(), info.ModTime(), info.Size())
 }
 
-// ServeContent comes from http.ServeContent, and made some adaptations for DownloadServer
-func ServeContent(
-	server DownloadServer, content io.ReadSeeker, contentType, name string, modTime time.Time, size int64,
+// serveContent comes from http.serveContent, and made some adaptations for DownloadServer
+func serveContent( //nolint:gocognit
+	server downloadServer, content io.ReadSeeker, contentType, name string, modTime time.Time, size int64,
```

Comment:
I recommend keeping `ServeContent` public, like `http.ServeContent`, so users can use io.ReadSeeker instead of just files